pkg/cloudprovider/aws/securitygroups.go

```diff
@@ -88,11 +95,37 @@ func (s *SecurityGroupProvider) getSecurityGroups(ctx context.Context, filters [
 		return nil, fmt.Errorf("describing security groups %+v, %w", filters, err)
 	}
 	s.cache.SetDefault(fmt.Sprint(hash), output.SecurityGroups)
-	logging.FromContext(ctx).Debugf("Discovered security groups: %s", s.securityGroupIds(output.SecurityGroups))
+	logging.FromContext(ctx).Debugf("Discovered security groups: %s", securityGroupIds(output.SecurityGroups))
 	return output.SecurityGroups, nil
 }
 
-func (s *SecurityGroupProvider) securityGroupIds(securityGroups []*ec2.SecurityGroup) []string {
+func filterClusterTaggedGroups(ctx context.Context, securityGroups []*ec2.SecurityGroup) []*ec2.SecurityGroup {
```

Comment:
No accessor to confine this to securitygroups instead of exposing to all of aws pkg?